### PR TITLE
Allow exceptions to be resolved from LogEntries

### PR DESCRIPTION
**Pull request by peppy**
Previously two log entries would be added for errors, making doing anything with them quite

### CODE REVIEW

**Comment by peppy** [comment]
This should be good to go now.

File: osu.Framework/Logging/Logger.cs [hunk 1]
```diff
@@ -118,12 +118,10 @@ public static void Error(Exception e, string description, string name, bool recu
 
         private static void error(Exception e, string description, LoggingTarget? target, string name, bool recursive)
         {
-            log($@"{description}", target, name, LogLevel.Error);
-            log(e.ToString(), target, name, LogLevel.Important);
```
Comment (smoogipoo) [inline osu.Framework/Logging/Logger.cs]: Why was this removed? We should display the exception text even if it was handled.

Comment (peppy) [reply osu.Framework/Logging/Logger.cs]: The reason for making this PR was to stop raw exception text from being considered a "log entry". Will look at adding this back by another means.

**Comment by smoogipoo** [comment]
Hmm with the following code:

```
[TestFixture]
public class TestCaseScratch : TestCase
{
    [Test]
    public void RunTest()
    {
        using (var host = new HeadlessGameHost())
        {
            host.Run(new TestGame());
        }
    }

    private class TestGame : Game
    {
        protected override void Update()
        {
            base.Update();

            throw new Exception("aa");
        }
    }
}
```

Two exceptions are thrown: the first is the exception in the sample. The second is an object disposed exception on the window. This second exception should not be allowed to happen, as it messes with debug as well (debugger incorrectly points to the disposal exception callstack).
